### PR TITLE
204 onboarding pallet get asset with finalising status

### PR DESCRIPTION
**Pull request by letodunc**
Updated Onboarding pallet with a new method to get the houses with the `FINALISING` status

### CODE REVIEW

File: pallets/onboarding/src/tests.rs [hunk 1]
```diff
@@ -443,3 +443,90 @@ fn get_finalised_houses_with_finalised_houses() {
...
+			NftColl::OFFICESTEST,
+			Some(price),
+			metadata1,
+			false
+		));
+
+		let collection_id = NftColl::OFFICESTEST.value();
+		let item_id = pallet_nft::ItemsCount::<Test>::get()[collection_id as usize] - 1;
+
+		// we simulate for the the presence of an finalised house by changing its status
```
Comment (cuteolaf) [inline pallets/onboarding/src/tests.rs:506]: // we simulate for the the presence of a finalised house by changing its status

File: pallets/onboarding/src/tests.rs [hunk 2]
```diff
@@ -443,3 +443,90 @@ fn get_finalised_houses_with_finalised_houses() {
...
+			metadata2,
+			false
+		));
+
+		// we check that the finalising house is correctly retrieved
+		let finalising_houses = OnboardingModule::get_finalising_houses();
+		assert_eq!(finalising_houses.len(), 1);
+
+		let house = finalising_houses[0].clone();
+		assert_eq!(house.2.status, AssetStatus::FINALISING,);
```
Comment (cuteolaf) [inline pallets/onboarding/src/tests.rs:529]: assert_eq!(house.2.status, AssetStatus::FINALISING);

**Review by cuteolaf — APPROVED** [review]

File: pallets/onboarding/src/functions.rs [hunk 3]
```diff
@@ -159,25 +159,38 @@ impl<T: Config> Pallet<T> {
...
 		<T as pallet_nft::Config>::NftItemId,
 		types::Asset<T>,
 	)> {
 		Houses::<T>::iter()
-			.filter(|val| val.2.status == types::AssetStatus::ONBOARDED)
-			.map(|elt| (elt.0, elt.1, elt.2))
+			.filter(|(_, _, house)| house.status == status)
+			.map(|(collection_id, item_id, house)| (collection_id, item_id, house))
 			.collect()
 	}
```
Comment (ndkazu) [inline pallets/onboarding/src/functions.rs:172]: Pallet Onboarding is also modified in PR #203 please make sure that the changes are coordinated

**Review by cuteolaf — APPROVED** [review]
looks good to me

**Review by ilhanu — APPROVED** [review]

**Pull request merged**